src/openfermion/measurements/_prony_test.py

```diff
@@ -0,0 +1,61 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+""" Tests for _prony.py"""
+import pytest
+from ._prony import prony
+import numpy
+
+
+class PronyTest:
+
+    def test_zeros(self):
+        signal = numpy.zeros(10)
+        amplitudes, phases = prony(signal)
+        assert (len(amplitudes) == 5)
+        assert (len(phases) == 5)
+        for j in range(5):
+            assert numpy.testing.assert_allclose(amplitudes[j], 0)
+            assert numpy.testing.assert_allclose(phases[j], 0)
+
+    def test_signal(self):
+        x_vec = numpy.linspace(0, 1, 11)
+        y_vec = (0.5 * numpy.exp(1j * x_vec * 3) +
+                 0.3 * numpy.exp(1j * x_vec * 5) +
+                 0.15 * numpy.exp(1j * x_vec * 1.5) +
+                 0.1 * numpy.exp(1j * x_vec * 4) +
+                 0.05 * numpy.exp(1j * x_vec * 1.2))
+        print(y_vec)
+        amplitudes, phases = prony(y_vec)
+        assert (len(amplitudes) == 5)
+        assert (len(phases) == 5)
+        for a, p in zip(amplitudes, phases):
+            print(a, numpy.angle(p))
+        assert numpy.testing.assert_allclose(amplitudes[0], 0.5, atol=0.001)
```

Comment:
The `assert` is unnecessary; `assert_allclose` already takes care of it.

Also, is 0.001 the smallest atol that this works for? That's a bit surprising to me.

Reply:
Good point, I was maybe being a bit too cautious there - this should be good to machine accuracy with the given data. I've changed the tol to 1e-9

Reply:
Scratch that, it seems that the error is about 1e-6 for the small signals instead of 1e-9. I think this is due to fixed point error in the least squares method (possibly in part because the matrix we invert is low rank). I'll try to see if I can improve on this over the next few weeks, but I think this might be an inherent issue with the estimator.